Source/Core/Core/PowerPC/CachedInterpreter/CachedInterpreter.cpp

```diff
@@ -118,7 +119,7 @@ void CachedInterpreter::Run()
     do
     {
       ExecuteOneBlock();
-    } while (PowerPC::ppcState.downcount > 0);
+    } while (PowerPC::ppcState.downcount > 0 && *state_ptr == CPU::State::Running);
```

Comment:
I'm not sure if the C++ code here is correct. Must a compiler still load the value from RAM even if state_ptr is marked as const?

Reply:
I think there is no guarantee unless it's volatile. `const volatile type *` is valid.

Reply:
Sorry, "on every iteration" is wrong here. It only needs to be checked if ExecuteOneBlock() may have changed it. So volatile isn't required here and the current code is well defined. Thanks for the help.